flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSinkV2.java

```diff
@@ -0,0 +1,333 @@
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.flink.source.BoundedTestSource;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.StructLikeSet;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestFlinkTableSinkV2 extends FlinkCatalogTestBase {
+  private static final String TABLE_NAME_V2 = "test_table_v2";
+  private static final int ROW_ID_POS = 0;
+  private static final int ROW_DATA_POS = 1;
+
+  private final boolean partitioned;
+
+  private final StreamExecutionEnvironment env;
+
+  private StreamTableEnvironment tEnv;
+  private Map<String, String> tableProps;
+
+  @Parameterized.Parameters(name = "CatalogName={0}, Format={1}, Partitioned={2}")
+  public static Iterable<Object[]> parameters() {
+    return ImmutableList.of(
+        new Object[] {"testhive", "avro", true},
+        new Object[] {"testhive", "avro", false},
+        new Object[] {"testhive", "parquet", true},
+        new Object[] {"testhive", "parquet", false},
+        new Object[] {"testhadoop", "avro", true},
+        new Object[] {"testhadoop", "avro", false},
+        new Object[] {"testhadoop", "parquet", true},
+        new Object[] {"testhadoop", "parquet", false}
+    );
+  }
+
+  public TestFlinkTableSinkV2(String catalogName, String format, Boolean partitioned) {
+    super(catalogName, new String[0]);
+    this.partitioned = partitioned;
+
+    this.env = StreamExecutionEnvironment.getExecutionEnvironment().enableCheckpointing(400);
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
+
+    this.tableProps = ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, format, TableProperties.FORMAT_VERSION, "2");
+  }
+
+  @Override
+  protected TableEnvironment getTableEnv() {
+    if (tEnv == null) {
+      synchronized (this) {
+        if (tEnv == null) {
+          tEnv = StreamTableEnvironment.create(env, EnvironmentSettings
+              .newInstance()
+              .useBlinkPlanner()
+              .inStreamingMode()
+              .build());
+        }
+      }
+    }
+    return tEnv;
+  }
+
+  @Before
+  public void before() {
+    super.before();
+    sql("CREATE DATABASE %s", flinkDatabase);
+    sql("USE CATALOG %s", catalogName);
+    sql("USE %s", DATABASE);
+  }
+
+  @After
+  public void clean() {
+    sql("DROP TABLE IF EXISTS %s.%s", flinkDatabase, TABLE_NAME_V2);
+    sql("DROP DATABASE IF EXISTS %s", flinkDatabase);
+    super.clean();
+  }
+
+  @Test
+  public void testSqlChangeLogOnIdKey() throws Exception {
```

Comment:
These tests look the same as the non-SQL tests. Is there a way to set up the test and expected data for both?

Reply:
I checked the source code of the flink cdc test class and found that it used docker to start a mysql service to get the cdc data. It seems that there really is no good way to get the cdc data in flink.